### ISSUE TITLE
Specify server machines

### ISSUE THREAD

**Issue by m1-s** [garnix-io/issues#100]
Hi,

just discovered garnix. If I understand correctly, garnix can deploy my nixos config on some server. Is there specification what hardware the server has? Can I upgrade/downgrade CPU or disk space depending on my needs?

Also I read that there is blue green deployment support: "On any commit, garnix spins up the new servers, waits for them to be ready, and then switches traffic to the new instances."

How does garnix detect when my server is up? Does it just wait until something is served on port 80? Can I configure my own conditions?

**Comment by jkarni** [comment]
> If I understand correctly, garnix can deploy my nixos config on some server. Is there specification what hardware the server has? Can I upgrade/downgrade CPU or disk space depending on my needs?

Yes, sorry, we should make that clearer on the website. The server that one gets for the 2 month trial, and for the duration of the individual place, is a 2 vCPU, 4 GB RAM intel VPS with 40 GB disk. You can get other machine types, though currently only by talking to us.

> How does garnix detect when my server is up? Does it just wait until something is served on port 80? Can I configure my own conditions?

When multi-user.target is ready. So yes: just use systemd! This is the same as what you'd get in a NixOS VM or doing nixos-rebuild switch

**Comment by m1-s** [comment]
Ok thanks for explaining. that sounds good. I can see some usecases of garnix for my clients. 

In the blue green deployment after routing traffic to the new machine, how does garnix know when it can terminate the old machine?

**Comment by jkarni** [comment]
> In the blue green deployment after routing traffic to the new machine, how does garnix know when it can terminate the old machine?

It sends a shutdown. Your program can take its time to shutdown, as long as your configure systems accordingly.
